### PR TITLE
UI Plugin: finished provider selection page and fixed a couple of css…

### PR DESCRIPTION
**Pull request by raymondz1**
## What this PR does / why we need it

1. created provider selection page.
2. fixed a couple of css issue.
3. navigate to progress page after deploy button is clicked.

## Which issue(s) this PR fixes

Fixes: https://github.com/vmware-tanzu/community-edition/issues/4193 https://github.com/vmware-tanzu/community-edition/issues/4294

### CODE REVIEW

**Review by swalner-vmware — APPROVED** [review]

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/validations/Validation.service.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ export const isValidIp = (arg: string | undefined) => {
     return regexPattern.test(arg.trim());
 };
 
+/**
+ * @method isValidClusterName decide if arg is a valid cluster name
+ * @return boolean
+ */
+
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/validations/Validation.service.ts:31]: remove extra line break

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/constants/NavRoutes.constants.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const NavRoutes = {
     AWS: 'aws',
 
     // temp routes to be refactored out
-    DEPLOY_PROGRESS: 'progress',
+    DEPLOY_PROGRESS: '/progress',
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/constants/NavRoutes.constants.tsx:14]: Please modify usage in ClusterAttributeStep, line 45 (navigateToProgress()), where the code is prepending '/' to that route

File: cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/management-cluster-settings-step/ManagementClusterSettings.tsx [hunk 3]
```diff
@@ -19,6 +19,10 @@ import { CdsButton } from '@cds/react/button';
 import { StepProps } from '../../../shared/components/wizard/Wizard';
 import { AwsStore } from '../../../state-management/stores/Store.aws';
 import './ManagementClusterSettings.scss';
+import { useNavigate } from 'react-router-dom';
```
[the remark this answers is not in the document]
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/management-cluster-settings-step/ManagementClusterSettings.tsx]: moved

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/validations/Validation.service.ts (hunk 1 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/shared/validations/Validation.service.ts]: removed

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/ManagementClusterLanding.tsx [hunk 4]
```diff
@@ -3,85 +3,79 @@ import React from 'react';
...
-                {
-                    cards.map((card, index) => {
+        <div
+            className="management-cluster-landing-container"
+            cds-layout="vertical gap:md col@sm:12 p:md grid"
+        >
+            <div cds-layout="col:8">
+                <h1 className="title" cds-text="body light" cds-layout="p-y:lg">Management Cluster</h1>
+                <p>
+                    Managed Clusters is a deployment model that features 1
```
[the remark this answers is not in the document]
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/ManagementClusterLanding.tsx]: fixed

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/constants/NavRoutes.constants.tsx (hunk 2 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/shared/constants/NavRoutes.constants.tsx]: fixed

File: cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/reducers/Ui.reducer.tsx [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
 import { Action } from '../../shared/types/types';
 
 interface UIState {
-    navExpanded: boolean,
-    wcCcRequiredExpanded: boolean,
-    wcCcOptionalExpanded: boolean,
+    navExpanded: boolean;
+    wcCcRequiredExpanded: boolean;
+    wcCcOptionalExpanded: boolean;
+    isInProgress: boolean;
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/reducers/Ui.reducer.tsx:14]: how about isDeployInProgress?

**Review by swalner-vmware — APPROVED** [review]

File: cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/reducers/Ui.reducer.tsx (hunk 5 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/reducers/Ui.reducer.tsx:14]: sounds good.

File: cli/cmd/plugin/ui/web/tanzu-ui/node-server/user_data/logs/access.log [hunk 6]
```diff
@@ -3441,3 +3441,7 @@
 2022-04-25T20:19:36.629Z ::ffff:127.0.0.1 - "GET /api/provider/aws/regions HTTP/1.1" 200 41 - 1.815 ms - "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.88 Safari/537.36"
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/node-server/user_data/logs/access.log:3441]: Raymond, sorry would you mind adding this file to .gitignore please?

Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/node-server/user_data/logs/access.log:3441]: will do it

**Review by miclettej — APPROVED** [review]

**Pull request merged**